src/main/java/twitterapp/src/services/TwitterAppService.java

```diff
@@ -0,0 +1,75 @@
+package twitterapp.src.services;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import twitter4j.Status;
+import twitter4j.Twitter;
+
+import java.util.List;
+
+
+public class TwitterAppService {
+    private static Logger log = (Logger) LoggerFactory.getLogger("myLogger");
+    static TwitterAppService service = null;
+    Twitter twitter;
+    int MAX_LENGTH = 280;
+
+    public static TwitterAppService getService() {
+        if (service == null) {
+            service = new TwitterAppService();
+        }
+        return service;
+    }
+
+    public void setTwitter(Twitter t) {
+        twitter = t;
+    }
+
+
+    public boolean testBadTweet(String tweet) throws IllegalArgumentException {
+
+        if (tweet.length() > MAX_LENGTH) {
+            log.warn("Tweet is too long, keep it within 280 characters");
+            throw new IllegalArgumentException("Tweet is too long, keep it within 280 characters");
+
+        } else if (tweet.length() == 0) {
+            log.warn("An empty tweet was entered");
+            throw new IllegalArgumentException("An empty tweet was entered");
+        } else {
+            return false;
+        }
+
+    }
+
+
+    public Status postTweet(String tweet) throws IllegalArgumentException {
```

Comment:
production code, if someone was to call your method, what errors would they get back compared to the resource layer call?